introspection-engine/introspection-engine-tests/tests/re_introspection/mod.rs

```diff
@@ -1674,3 +1674,51 @@ async fn re_introspecting_ignore(api: &TestApi) -> crate::TestResult {
 
     Ok(())
 }
+
+#[test_each_connector]
+async fn do_not_try_to_keep_custom_many_to_many_self_relation_names(api: &TestApi) -> crate::TestResult {
+    //we do not have enough information to correctly assign which field should point to column A in the
+    //join table and which one to B
+    //upon table creation this is dependant on lexicographic order of the names of the fields, but we
+    //cannot be sure that users keep the order the same when renaming. worst case would be we accidentally
+    //switch the directions when reintrospecting.
+    //the generated names are also not helpful though, but at least they don't give a false sense of correctness -.-
```

Comment:
It's weird we (or users) didn't bump into this issue earlie.r

Reply:
Well, luckily reintrospection was crashing on that case before ;-)